### PR TITLE
[FIX] website_sale_collect: hide 'Send to Shipper' button for 'in_store' delivery type

### PR DESCRIPTION
**Pull request by aamo-odoo**
This issue occurs when a delivery order is created with the ``Carrier`` set to ``Pick up in Store``. After saving and validating the order, click the ``Send to Shipper`` button.

Steps to reproduce:
---
- Install ``website_sale_collect`` module
- Create a new ``Delivery Order`` and in ``Additional Info`` select ``Carrier`` as ``Pick up in Store`` > Save and Validate
- Click on the ``Send to Shipper`` button

Traceback:
--- 
``TypeError: 'NoneType' object is not subscriptable``

This commit resolves the issue by adding an XPath to the view to hide the ``Send to Shipper`` button for orders with the delivery type set to ``in_store``.

sentry-6040319897

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/website_sale_collect/__manifest__.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
         'views/delivery_carrier_views.xml',
         'views/delivery_form_templates.xml',
+        'views/delivery_pick_up_in_store_views.xml',
```
Comment (vchu-odoo) [inline addons/website_sale_collect/__manifest__.py:19]: Could you please rename the file?
```suggestion
        'views/stock_picking_views.xml',
```

File: addons/website_sale_collect/views/delivery_pick_up_in_store_views.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_picking_withcarrier_out_form_inherit_puis" model="ir.ui.view">
```
Comment (vchu-odoo) [inline addons/website_sale_collect/views/delivery_pick_up_in_store_views.xml:3]:
```suggestion
    <record id="stock_picking_form" model="ir.ui.view">
```

File: addons/website_sale_collect/views/delivery_pick_up_in_store_views.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+        <field name="name">stock.picking.form.inherit.puis</field>
```
Comment (vchu-odoo) [inline addons/website_sale_collect/views/delivery_pick_up_in_store_views.xml:4]:
```suggestion
        <field name="name">stock.picking.form</field>
```

**Review by vchu-odoo — CHANGES_REQUESTED** [review]
Thank you for your fix!

Just some small suggestions.

**Comment by aamo-odoo** [comment]
Hey @vchu-odoo 

Requested changes are made.
You can now review this. 

Thanks! :)

**Review by vchu-odoo — APPROVED** [review]
Thank you for the changes!

LGTM :+1:

File: addons/website_sale_collect/views/stock_picking_views.xml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="stock_picking_form" model="ir.ui.view">
+        <field name="name">stock.picking.form</field>
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock_delivery.view_picking_withcarrier_out_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//button[@name='send_to_shipper']" position="attributes">
```
Comment (AntoineVDV) [inline addons/website_sale_collect/views/stock_picking_views.xml:8]:
```suggestion
            <button name="send_to_shipper" position="attributes">
```

**Review by AntoineVDV — APPROVED** [review]
@odoo/robodoo delegate=vchu-odoo

Although `ci/template` is red, the change is fine for stable. However, it will require an upgrade exception from the runbot team to be merged.

**Comment by vchu-odoo** [comment]
@robodoo r+

**Comment by aamo-odoo** [comment]
Hey @vchu-odoo 

We need an exception from runbot team

**Comment by d-fence** [comment]
Upgrade exception [#463](https://runbot.odoo.com/odoo/runbot.upgrade.exception/463?menu_id=405) added.

Waiting the forward-port of this pr, the exception will be applied on all builds, and create an inconsistent state for migrations.
**Please forward port this asap up to master without change.** Exception should be forward-ported before the end of the week.

PRs adding fields/models should be merged at the beginning of the week and all the forward ports should be merged in the same week.

If you need to apply any change before it reaches master, please notify runbot team.

Details:
```
view:website_sale_collect.stock_picking_form
```
cc @KangOl @nseinlet @aj-fuentes

**Pull request closed without merging**